dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -678,65 +680,113 @@ private void deleteTreesForInodes(List<String> inodes) throws DotDataException {
-        //this string is converted to Uppercase.This does
-        //not cause a problem with the other databases
+        final Date date = calendar.getTime();
         DotConnect dc = new DotConnect();
-
-        String countSQL = ("select count(*) as count from contentlet");
+        final String countSQL = "select count(*) as count from contentlet";
         dc.setSQL(countSQL);
         List<Map<String, String>> result = dc.loadResults();
-        int before = Integer.parseInt(result.get(0).get("count"));
-
-        String deleteContentletSQL = "delete from contentlet where identifier<>'SYSTEM_HOST' and mod_date < ? " +
-        "and not exists (select * from contentlet_version_info where working_inode=contentlet.inode or live_inode=contentlet.inode)";
-        dc.setSQL(deleteContentletSQL);
+        final int before = Integer.parseInt(result.get(0).get("count"));
+        dc = new DotConnect();
+        dc.setSQL("SELECT inode FROM contentlet WHERE identifier <> 'SYSTEM_HOST' AND mod_date < ? AND NOT EXISTS " +
+                "(SELECT working_inode, live_inode FROM contentlet_version_info WHERE working_inode = contentlet.inode OR " +
+                "live_inode = contentlet.inode)");
         dc.addParam(date);
-        dc.loadResult();
-
-        String deleteOrphanInodes="delete from inode where type='contentlet' and idate < ? and inode not in (select inode from contentlet)";
-        dc.setSQL(deleteOrphanInodes);
-        dc.addParam(date);
-        dc.loadResult();
-
+        result = dc.loadResults();
+        int oldInodesCount = result.size();
+        if (oldInodesCount > 0) {
+            final List<String> inodeList = result.stream().map(row -> row.get("inode")).collect(Collectors.toList());
+            deleteContentData(inodeList);
+        }
+        dc = new DotConnect();
         dc.setSQL(countSQL);
         result = dc.loadResults();
-        int after = Integer.parseInt(result.get(0).get("count"));
+        final int after = Integer.parseInt(result.get(0).get("count"));
+        final int deleted = before - after;
+        if (deleted > 0) {
+            deleteOrphanedBinaryFiles();
+        }
+        return deleted;
+    }
 
-        int deleted=before - after;
+    /**
+     * Deletes the content data associated to the specified list of Inodes. Based on such a list, the {@code contentlet}
+     * will be cleaned up as well.
+     *
+     * @param inodeList The list of Inodes that will be deleted.
+     *
+     * @throws DotDataException An error occurred when interacting with the data source.
+     */
+    private void deleteContentData(final List<String> inodeList) throws DotDataException {
+        final int splitAt = 100;
+        // Split all records into lists of size 'truncateAt'
+        final List<List<String>> inodesToDelete = Lists.partition(inodeList, splitAt);
+        final List<String> queries = Lists.newArrayList("DELETE FROM contentlet WHERE inode IN (?)",
```

Comment:
You can use a temp table instead and do a `DELETE JOIN` instead of having to use an `IN`. 
Basically insert results of the previous suggested query (UNION) into a temp table, and join against that temp table for doing the two DELETEs